entity-registry/src/main/java/com/linkedin/metadata/models/EntitySpec.java

```diff
@@ -40,15 +47,64 @@ default List<SearchableFieldSpec> getSearchableFieldSpecs() {
   }
 
   default Map<String, Set<SearchableFieldSpec>> getSearchableFieldSpecMap() {
-    return getSearchableFieldSpecs().stream()
-        .collect(
-            Collectors.toMap(
-                searchableFieldSpec -> searchableFieldSpec.getSearchableAnnotation().getFieldName(),
-                searchableFieldSpec -> new HashSet<>(Collections.singleton(searchableFieldSpec)),
-                (set1, set2) -> {
-                  set1.addAll(set2);
-                  return set1;
-                }));
+    // Get additional fields and mint SearchableFieldSpecs for them
+    Map<String, Set<SearchableFieldSpec>> fieldSpecMap = new HashMap<>();
+    for (SearchableFieldSpec fieldSpec : getSearchableFieldSpecs()) {
+      SearchableAnnotation searchableAnnotation = fieldSpec.getSearchableAnnotation();
+      if (searchableAnnotation.getNumValuesFieldName().isPresent()) {
+        String fieldName = searchableAnnotation.getNumValuesFieldName().get();
+        SearchableFieldSpec numValuesField =
+            getSearchableFieldSpec(
+                fieldName, SearchableAnnotation.FieldType.COUNT, new LongDataSchema());
+        Set<SearchableFieldSpec> fieldSet = new HashSet<>();
+        fieldSet.add(numValuesField);
+        fieldSpecMap.put(fieldName, fieldSet);
+      }
+      if (searchableAnnotation.getHasValuesFieldName().isPresent()) {
```

Comment:
@jjoyce0510 and I talked about this method being undesirable since it is creating a mostly empty annotation and an annotation that doesn't really reflect the expectations of the boolean field.

The thought was to create/generate a field map with a simpler value than a full scale annotation object.

Reply:
Sure, it really only needs a couple of the fields so that makes sense.

Reply:
SearchableAnnotation should directly model the annotation inside the PDL files. That is the charter for that object..

If we need another intermediary abstraction to represent the _Actual_ physical index fields that get minted as a result of a SearchableAnnotation, we should create it. Something like SearchableField or something. It can store a pointer to the origin SearchableAnnotation if required.

Reply:
That's what SearchableFieldSpec already is, we just directly access the annotation object for all the information we use instead of duplicating it.